rkt/images.go

```diff
@@ -402,6 +405,20 @@ func (f *fetcher) fetch(aciURL, ascURL string, ascFile *os.File) (*openpgp.Entit
 		defer ascFile.Close()
 	}
 
+	// check if the identity used by the signature is in the store before a
+	// possibly expensive download. This is only an optimization and it's
+	// ok to skip the test if the signature will downloaded later.
+	if !retrySignature && f.ks != nil && appName != "" {
+		if _, err := ascFile.Seek(0, 0); err != nil {
+			return nil, nil, fmt.Errorf("error seeking signature file: %v", err)
+		}
+		if entity, err = f.ks.CheckSignature(appName, bytes.NewBuffer([]byte{}), ascFile); err != nil {
+			if _, ok := err.(pgperrors.SignatureError); !ok {
+				return nil, nil, err
```

Comment:
I'm a little concerned about obviousness - what's the actual error message that gets printed here?